src/sssom/util.py

```diff
@@ -165,7 +165,15 @@ def from_mapping_set_document(cls, doc: MappingSetDocument) -> "MappingSetDataFr
             # For pandas < 2.0.0, call 'infer_objects()' without any parameters
             df = df.infer_objects()
         # remove columns where all values are blank.
+
+        # Context https://github.com/pandas-dev/pandas/issues/57734
+        try:
+            pd.set_option("future.no_silent_downcasting", True)
```

Comment:
I would rather put that somewhere when the library is initialized, so that we can be sure the option is enabled everywhere at all times and not only when the code path goes through this particular function.

Reply:
I added it to sssom/__init__.py, not sure this is the right thing to do (can this effect other libraries that might not want that option to be set for some reason?)

Reply:
I would personally be fine having it in `sssom/init.py`. This is forcing the behaviour to apply globally, which I believe is a _good_ thing.

That being said, it is true that this could have cascading effects for people who are using `sssom` as a library rather than as a command-line tool (people who import `sssom` in their own Python projects, rather than calling `sssom-py` from the command line), if they also happen to be using `pandas` in the rest of their code (no problem if their only use of `pandas` is through `sssom`).

If you’re not comfortable forcing the “no downcasting” behaviour to all potential users of the `sssom` Python module¹, I can think of two options:

(A) Setting the option somewhere at the beginning of the `main` function in `sssom/cli.py`. This means the option will be enabled globally in all of `sssom` _when it is used as a CLI tool_, but not when it is used as an imported module in another Python project.

It will then be up to the people who imports `sssom` to decide whether they themselves want to enable the “no downcasting” behaviour, by setting (or not) the option.

(B) Enabling the option within `sssom`, but only where we need it and only for the duration we need it. That is, we enable it before calling `replace` here, and disable it afterwards.

A context manager would come in handy here:

```python
class pandas_no_silent_downcasting():

    def __init__(self):
        try:
            self._already_set = pd.get_option('future.no_silent_downcasting')
            self._supported = True
        except pd.errors.OptionError:
            self._supported = False

    def __enter__(self):
        if self._supported and not self._already_set:
            # Entering the context, set the option
            pd.set_option('future.no_silent_downcasting', True)

    def __exit__(self, type, value, tb):
        if self._supported and not self._already_set:
            # Leaving the context, unset the option
            pd.set_option('future.no_silent_downcasting', False)
```

It could then be used whenever we have to perform an operation and need to locally make sure the option (if it is available and if it is not already set) is enabled:

```python
with pandas_no_silent_downcasting():
    df.replace(...)
```

This is equivalent to:

```python
pd.set_option('future.no_silent_downcasting', True)
df.replace(...)
pd.set_option('future.no_silent_downcasting', False)
```

except that it is more “pythonic” and it deals gracefully with the possible absence of that option or the possibility that the option has already been enabled elsewhere.

---
¹ But it should be noted that this behaviour _will_ be forced upon them sooner or later anyway, since it will be the only behaviour available in Pandas 3.0… Better for those people to start dealing with it now.